File: 19-remove-nth-node-from-end-of-list/19-remove-nth-node-from-end-of-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    void solve(ListNode* &head, ListNode* prev,int &cnt,int n){
        if(!head)return;
        solve(head->next,head,cnt,n);
        if(cnt!=n)cnt++;
        else{
            if(head and prev) prev->next = head->next;
            else if(prev) prev->next = NULL;
            else if(head) head = head->next;
            cnt++;
        }
    }
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        if(!head->next)return NULL;
        int cnt=1;
        solve(head, NULL,cnt,n);
        return head;
    }
};

File: 19-remove-nth-node-from-end-of-list/19-remove-nth-node-from-end-of-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} n
 * @return {ListNode}
 */
var removeNthFromEnd = function(head, n) {
    let start = new ListNode();
    start.next = head;
    let slow = start, fast = start;
    while(n--)fast = fast.next;
    while(fast.next){
        fast = fast.next;
        slow = slow.next;
    }
    slow.next = slow.next.next;
    return start.next;
};